src/qspdlog_model.cpp

```diff
@@ -145,3 +160,39 @@ QVariant QSpdLogModel::headerData(
 
     return QVariant();
 }
+void QSpdLogModel::setForegroundMapping(
+    const std::string& loggerName, std::optional<QBrush> brush
+)
+{
+    if(brush.has_value())
+        _foregroundMappings[loggerName] = brush.value();
+    else if(_foregroundMappings.contains(loggerName))
+        _foregroundMappings.erase(loggerName);
+}
+std::optional<QBrush> QSpdLogModel::getForegroundMapping(
+    const std::string& loggerName
+) const
+{
+    if(_foregroundMappings.contains(loggerName))
+        return _foregroundMappings.at(loggerName);
+
+    return std::nullopt;
+}
+void QSpdLogModel::setBackgroundMapping(
+    const std::string& loggerName, std::optional<QBrush> brush
+)
+{
+    if(brush.has_value())
+        _backgroundMappings[loggerName] = brush.value();
+    else if(_backgroundMappings.contains(loggerName))
+        _backgroundMappings.erase(loggerName);
+}
+std::optional<QBrush> QSpdLogModel::getBackgroundMapping(
+    const std::string& loggerName
```

Comment:
`const std::string&` -> `std::string_view`
```suggestion
    std::string_view loggerName
```